fv3core/pace/fv3core/stencils/fv_subgridz.py

```diff
@@ -783,12 +782,17 @@ def __init__(
         n_sponge: int,
         hydrostatic: bool,
     ):
-        assert not hydrostatic, "Hydrostatic not implemented for fv_subgridz"
+        if hydrostatic:
+            raise NotImplementedError(
+                "DryConvectiveAdjustment (fv_subgridz):"
+                " Hydrostatic is not implemented"
+            )
         grid_indexing = stencil_factory.grid_indexing
         self._k_sponge = n_sponge
-        if self._k_sponge is not None:
-            if self._k_sponge < 3:
-                return
+        if self._k_sponge is not None and self._k_sponge < 3:
+            raise NotImplementedError(
```

Comment:
This should be a value error not a not implemented error, the model shouldn't have nsponge < 3

Reply:
Done